apps/dashboard/src/app/api/testnet-faucet/claim/route.ts

```diff
@@ -0,0 +1,83 @@
+import { getIpAddress } from "lib/ip";
+import { cacheExists, cacheSet } from "lib/redis";
+import { NextResponse } from "next/server";
+import { ZERO_ADDRESS } from "thirdweb";
+
+interface RequestTestnetFundsPayload {
+  chainId: number;
+  toAddress: string;
+  amount: string;
+}
+
+// Note: This handler cannot use "edge" runtime because of Redis usage.
+export const POST = async (req: Request) => {
+  const requestBody = (await req.json()) as RequestTestnetFundsPayload;
+  const { chainId, toAddress, amount } = requestBody;
+  if (Number.isNaN(chainId)) {
+    throw new Error("Invalid chain ID.");
+  }
+
+  const ipAddress = getIpAddress();
+  const cacheKey = `testnet-faucet:${chainId}:${ipAddress}`;
+
+  // Assert 1 request per IP/chain every 24 hours.
+  if (await cacheExists(cacheKey)) {
+    return NextResponse.json(
+      { error: "Already requested funds on this chain in the past 24 hours." },
+      { status: 429 },
+    );
+  }
```

Comment:
*have* to set the cache right after this happens, otherwise I can call this 1000 times in < [request time] and get the funds 1000x

Reply:
you're right! done